### DISCUSSION TITLE
wix5 burn bootstrap exe installation, how to make the key in the registry fixed

### DISCUSSION THREAD

**Discussion by lc5259** [wixtoolset/issues#8648] [Questions] [answered]
I am installing using the exe generated by the burn bundle and find that every time I install, In the computer \HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall directory is a random guid each time. I want to fix this id. What do I do

**Comment by bevanweiss** [comment]
https://learn.microsoft.com/en-us/windows/win32/msi/uninstall-registry-key

Depending on how your authoring is done, it's possible that the GUID shown is a unique Product ID for your MSI.
If you currently have this set to "*" it will be automatically re-generated each time, which is why it is 'random'
https://stackoverflow.com/questions/10330534/in-wix-where-is-the-productcode-specified
PS: You'll need to convert the Wix3 from Stackoverflow to Wix5 syntax ([Package](https://wixtoolset.org/docs/schema/wxs/package/))

**Comment by lc5259** [comment, marked as answer]
You are talking about the setting of the msi package product id. I want to set the id of the Bundle.
<Bundle Name="XXXX" Version="2.0.2407.47228" Manufacturer="XXXX" UpgradeCode="{F158FCFD-C614-7118-9DF9-9AD63AC3A270}" IconSourceFile="xXX.ico">

**Reply by lc5259** [reply]
<Bundle Name="XXXX" Version="2.0.2407.47228" Manufacturer="XXXX" UpgradeCode="{F158FCFD-C614-7118-9DF9-9AD63AC3A270}" IconSourceFile="xXX.ico">

**Reply by bevanweiss** [reply]
Show us a screenshot of the registry values under this random GUID key.
I'm very much expecting it to be an MSI entry, which your burn bundle is deploying, and hence this GUID is coming from the MSI that you are also authoring (since it's changing each time you build...).

**Reply by lc5259** [reply]
![image](https://github.com/user-attachments/assets/6b0a45eb-fc39-4079-9a0d-e612484e471c)
![image](https://github.com/user-attachments/assets/e96fe24c-03f9-47dd-b5a0-e1089920c746)
 第一张图是msi 包在注册表中信息。第二张是exe包在注册表中的信息。每次exe包安装时候，guid都是随机的

**Reply by bevanweiss** [reply]
You haven't configured the ProviderKey to be non-unique
![image](https://github.com/user-attachments/assets/27aa34c7-b218-4f28-959d-be5233e08cd9)

https://wixtoolset.org/docs/schema/wxs/bundle/
**ProviderKey (String)** : Optional attribute to explicitly author the provider key for the entire bundle.This provider key is designed to persist throughout compatible upgrades so that dependent bundles do not have to be reinstalled and will not prevent your product from being upgraded. If this attribute is not authored, the value is the automatically-generated bundle ID and will not automatically support upgrades.
